tensorflow_lite_support/cc/task/vision/landmark_detector.cc

```diff
@@ -0,0 +1,157 @@
+        StatusCode::kInvalidArgument,
+        absl::StrFormat("Missing mandatory `model_file` field in `base_options`"),
+        TfLiteSupportStatus::kInvalidArgumentError);
+  }
+  RETURN_IF_ERROR(landmark_detector->Init(std::move(options_copy)));
+
+  return landmark_detector;
+}
+
+/* static */
+absl::Status LandmarkDetector::SanityCheckOptions(
+    const LandmarkDetectorOptions& options) {
+  // Nothing to check
+  return absl::OkStatus();
+}
+/* static */
+absl::Status SanityCheckOutputTensors(
+    const std::vector<const TfLiteTensor*>& output_tensors) {
+  if (output_tensors.size() != 1) {
+    return CreateStatusWithPayload(
+        StatusCode::kInternal,
+        absl::StrFormat("Expected 1 output tensors, found %d",
+                        output_tensors.size()));
+  }
+  return absl::OkStatus();
+}
+
+absl::Status LandmarkDetector::Init(
+    std::unique_ptr<LandmarkDetectorOptions> options) {
+  // Set options.
+  options_ = std::move(options);
+
+  // Perform pre-initialization actions (by default, sets the process engine for
+  // image pre-processing to kLibyuv as a sane default).
+  RETURN_IF_ERROR(PreInit());
+
+  // Sanity check and set inputs and outputs.
+  RETURN_IF_ERROR(CheckAndSetInputs());
+
+  return absl::OkStatus();
+}
+
+absl::Status LandmarkDetector::PreInit() {
+  SetProcessEngine(FrameBufferUtils::ProcessEngine::kLibyuv);
+  return absl::OkStatus();
+}
+
+StatusOr<LandmarkResult> LandmarkDetector::Detect(
+    const FrameBuffer& frame_buffer) {
+  BoundingBox roi;
+  roi.set_width(frame_buffer.dimension().width);
+  roi.set_height(frame_buffer.dimension().height);
+  return Detect(frame_buffer, roi);
+}
+
+StatusOr<LandmarkResult> LandmarkDetector::Detect(
+    const FrameBuffer& frame_buffer, const BoundingBox& roi) {
+  return InferWithFallback(frame_buffer, roi);
+}
+
+StatusOr<LandmarkResult> LandmarkDetector::Postprocess(
+    const std::vector<const TfLiteTensor*>& output_tensors,
+    const FrameBuffer& /*frame_buffer*/, const BoundingBox& /*roi*/) {
+  // Most of the checks here should never happen, as outputs have been validated
+  // at construction time. Checking nonetheless and returning internal errors if
+  // something bad happens.
+  RETURN_IF_ERROR(SanityCheckOutputTensors(output_tensors));
+
+  // Get number of keypoints.
+  const int num_keypoints = output_tensors[0]->dims->data[2];
+  const float* outputs = AssertAndReturnTypedTensor<float>(output_tensors[0]);
+
+  LandmarkResult result;
+
+  for (int i = 0; i < num_keypoints; ++i) {
+    Landmark* landmarks = result.add_landmarks();
+    // Set Scores
+    landmarks->set_score(outputs[3 * i + 2]);
+    // Set y coordinates
+    landmarks->add_position(0);
+    landmarks->set_position(0, outputs[3 * i + 0]);
+    // Set x coordinates
+    landmarks->add_position(1);
+    landmarks->set_position(1, outputs[3 * i + 1]);
+  }
+
+  return result;
+}
+
+}  // namespace vision
+}  // namespace task
+}  // namespace tflite
```

Comment:
All files should end with a new blank line.

Reply:
Means? It looks the same as all other libraries.

Reply:
Adding a `\n` after `} // namespace tflite`.

Reply:
ok, thanks!